### ISSUE TITLE
Unit test failures on Windows

### ISSUE THREAD

**Issue by eliotst** [move-coop/parsons#282]
A user saw test failures when running `pytest` on Windows against the Parsons test package. Specifically, they saw 8 failures in the `test.test_smtp.TestSMTP` and `test.test_gmail.TestGmail`.

**Comment by sarah-russell12** [comment]
I was the user who saw test failures.  I have a few screenshots of what the primary problems are.  The test failures in TestGmail all seem to be in the tests for adding attachments to emails, and essentially all looked like this:
![TestGmail-test_create_message_attachments-failed](https://user-images.githubusercontent.com/10925364/85423765-181ac580-b545-11ea-9a23-bcfb61f376fd.png)
The test failure in from TestSMP was this:
![TestSMP-failed-test](https://user-images.githubusercontent.com/10925364/85423897-3f719280-b545-11ea-8dde-07249d327bfe.png)

**Comment by sarah-russell12** [comment]
I also saw this error catching message in between most of the test failure readouts, though that may just be a problem on my end:
![TestGmail-stderr-output](https://user-images.githubusercontent.com/10925364/85424131-88c1e200-b545-11ea-8cb5-967749ee6f82.png)

**Comment by dannyboy15** [comment]
For the Gmail test failures it appears to be an issue with the new lines characters `\n`. I think we may have to change how we test the those specific components are there.

**Comment by sarah-russell12** [comment]
As an update, I had to do a clean install of Windows recently.  To get parsons back on my computer, I followed all the documentation for forking, cloning, and then setting up a virtual environment for installing parsons dependencies.  After doing that, I ran the tests again and I got only one test failure in test_create_message_attachments in TestGmail.
![TestGmail-test_create_message_attachments-Failure](https://user-images.githubusercontent.com/10925364/91481526-93b12f80-e872-11ea-9218-6bf2f7c62138.png)
Here is the Log Call I got after it as well.
![Captured-Log-Call](https://user-images.githubusercontent.com/10925364/91481599-b2172b00-e872-11ea-8463-f90a5406295b.png)

According to Elliot, it is still an issue with newlines on Windows.

**Comment by sarah-russell12** [comment]
I looked further into the issue and it is an issue with Windows newlines.  I first created a function in TestGmail that compares the output of the `Gmail._create_message_attachments` encryption and decryption against the text of the file the test opens for comparison.
![b64_encoding_comparison](https://user-images.githubusercontent.com/10925364/91902537-e2ded200-ec6f-11ea-8792-a37ee7d6f242.png)

Then after some digging in the code and the Python docs I learned some things: So the way Python's `open` works is that there are different modes.  `SendMail._create_message_attachments` appears to use read-binary mode when working with text files.  When reading in text mode, by default, all forms of newlines are automatically converted to \n when read, including the Windows newline \n\r.  `open` in binary mode doesn't do this, it retains the \r.

I found this difference by creating my own script that opens the loremipsum.txt file both in binary mode and in text mode, and then I converted the string from text mode into a byte stream.  Then for good measure, I followed the base64 encryption and decryption steps from `SendMail._create_message_attachments` and `TestGmail.test_create_message_attachments` as best I could and then compared the output the same way I did above.  And the results are similar.
![TestOpen-output](https://user-images.githubusercontent.com/10925364/91902624-030e9100-ec70-11ea-82d3-942a375f2af7.png)

So I have two ideas for possible solutions:
1. The quick and dirty way of dealing with this could be to just have a separate file that has the output of the encoding and decoding on Windows, and choose which to use in the test based on your operating system
2. The other way would be to alter the way we handle text files: by first opening and reading them in text mode, and then converting the text to a byte stream.  That way the carriage return byte is excluded from the byte stream no matter which operating system you're on (hopefully).
